inference-engine/ie_bridges/python/tests/test_InferRequest.py

```diff
@@ -350,7 +351,7 @@ def execute(self, input_data):
             self.cv.release()
             status = self.request.wait(ie.WaitMode.RESULT_READY)
             assert status == ie.StatusCode.OK
-            assert self.status_code == ie.StatusCode.OK
+            assert self.status_code == ie.StatusCode.RESULT_NOT_READY
```

Comment:
It looks strange from API level perspective. User calls wait with wait mode  == `RESULT_READY`, but receives `RESULT_NOT_READY`

Reply:
I am curious about it, why we get `RESULT_NOT_READY`? I understand the idea behind it is to capture status inside callback which will be equal to `RESULT_NOT_READY`, and only after returning to python we can obtain `OK`. However this could render previous callbacks' "policy" absolutely obsolete. The behavior was even present around March of the current year:
https://github.com/openvinotoolkit/openvino/blob/6478f1742a62310bd4b65a649a08c6c1b7d7949b/inference-engine/ie_bridges/python/sample/classification_sample_async/classification_sample_async.py#L24-L29

I totally understand an idea to be able to call wait inside callback itself. However why we are still passing "the same" status code as a parameter? To my knowledge this `statusCode` would be equal to `OK` as cpp implementation marks it this way as we are able to enter a callback and not to confuse users with different names of callback's exit codes. The parameter is here:
https://github.com/openvinotoolkit/openvino/blob/809bb8af2a4347ae22aba0a99bb248aba9c65b0c/inference-engine/ie_bridges/python/tests/test_InferRequest.py#L341

This needs a follow-up (in my opinion) as it requires:
1) either reconsidering a behavior of the callbacks in python and picking of the one "true" way of obtaining the status inside the callback
2) or changing c++ core code to apply aligned behavior in both APIs.

@nkogteva @akuporos what are your views on that?
@AlexeyLebedev1 could you elaborate more on this change?

Reply:
@jiwaszki From my point of view, we can't disallow the user to call wait inside callback and vice versa we should not force to do it just to check the status code. I agree that it is strange if status codes received in these two ways would be different. At least it should be documented.

But anyway the behavior should be aligned with C++. cc @ilya-lavrenov

Reply:
The main point of these changes that InferRequest.wait(RESULT_READY) must block the thread until infer_callback function completes, as in c++ api. When we call wait inside callback with no-wait status, we get RESULT_NOT_READY, also like in c++ api and it can confuse user, i agree, but if we want to wait for infer_callback maybe it makes sense.

Reply:
One more note: as far as I know user can access output blobs inside callback like in [demo pipeline](https://github.com/openvinotoolkit/open_model_zoo/blob/b1ff98b64a6222cf6b5f3838dc0271422250de95/demos/common/python/pipelines/async_pipeline.py#L102-L107). Thus it makes sense to provide the correct status code which informs of present state - which is `OK`. @ilya-lavrenov is it possible to obtain results in C++ API's callbacks as well? If yes, we should consider changes to C++ API to return `OK` in callback or some new status that is clearer about output blobs status instead of focusing on InferRequest's "life".

If we are going to align code to C++ without any changes then it should be properly documented and reflected on the python side.

Reply:
How i knew from @ilya-lavrenov , c++ api supposses that callback is the last stage of inference, that's why `InferRequest.wait(0)` inside callback returns RESULT_NOT_READY.  I don't see why we should worry about this status code https://github.com/openvinotoolkit/openvino/blob/f79d64977a61dbe5cf65cbe7e8e279ac14dffaee/inference-engine/ie_bridges/python/tests/test_InferRequest.py#L341 , if we already got status code, as an argument, https://github.com/openvinotoolkit/openvino/blob/f79d64977a61dbe5cf65cbe7e8e279ac14dffaee/inference-engine/ie_bridges/python/tests/test_InferRequest.py#L340, which is OK, otherway we would get an exception https://github.com/openvinotoolkit/openvino/blob/f79d64977a61dbe5cf65cbe7e8e279ac14dffaee/inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api_impl.cpp#L538-L545 , so we can safety access output blobs inside callback.